### PR TITLE
Fixed Android AzCore Utils.cpp path

### PR DESCRIPTION
**Pull request by moraaar**
Signed-off-by: moraaar <moraaar@amazon.com>

### CODE REVIEW

**Review by amzn-sj — APPROVED** [review]

**Review by rgba16f — APPROVED** [review]

**Review by AMZN-ScottR — APPROVED** [review]

File: Code/Framework/AzCore/Platform/Android/platform_android_files.cmake [hunk 1]
```diff
@@ -86,7 +86,7 @@ set(FILES
 )
 if (LY_TEST_PROJECT)
     ly_add_source_properties(
-        SOURCES AzCore/Android/Utils.cpp
+        SOURCES Platform/Android/AzCore/Android/Utils.cpp
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzCore/Platform/Android/platform_android_files.cmake:89]: Actually this should be relative to the current directory.
If anything what should be used here is the `${CMAKE_CURRENT_LIST_DIR}` to get the current directory of this cmake file

Comment (moraaar) [reply Code/Framework/AzCore/Platform/Android/platform_android_files.cmake:89]: Like this?

````
SOURCES ${CMAKE_CURRENT_LIST_DIR}/AzCore/Android/Utils.cpp
````

Comment (lumberyard-employee-dm) [reply Code/Framework/AzCore/Platform/Android/platform_android_files.cmake:89]: Yes, that should work

Comment (moraaar) [reply Code/Framework/AzCore/Platform/Android/platform_android_files.cmake:89]: Cool, thank. I'll test it locally and update the PR if it passes.

Comment (moraaar) [reply Code/Framework/AzCore/Platform/Android/platform_android_files.cmake]: It worked, updating PR.

**Pull request merged**